tensorbay/cli/log.py

```diff
@@ -24,32 +28,136 @@
 """
 
 
+class LogCommits:
+    """This class defines the structure of logging commits.
+
+    Arguments:
+        all_commits: The all commits PagingList from valid branches.
+
+    """
+
+    def __init__(self, all_commits: List[PagingList[Commit]]):
+        # Sorted commits paging list by date.
+        self.sorted_commits: List[PagingList[Commit]] = sorted(
+            all_commits, key=lambda x: x[0].committer.date
+        )
+        self.keys = [commits[0].committer.date for commits in self.sorted_commits]
+
+    def sort(self) -> None:
+        """Sort commits paging list by commit date."""
+        length = len(self.sorted_commits)
+        # Only one branch exist.
+        if length == 1:
+            return
+
+        # Binary insert.
+        commits = self.sorted_commits.pop()
+        self.keys.pop()
```

Comment:
`date = self.keys.pop()`?

Reply:
The pop date is useless, the new commit's date is useful